src/lib/logical_query_plan/aggregate_node.cpp

```diff
@@ -157,6 +157,23 @@ UniqueColumnCombinations AggregateNode::unique_column_combinations() const {
   return unique_column_combinations;
 }
 
+OrderDependencies AggregateNode::order_dependencies() const {
+  auto order_dependencies = OrderDependencies{};
+
+  // Similarly to UCCs, forward ODs if all expressions are part of the GROUP-BY expressions.
+  const auto& input_order_dependencies = left_input()->order_dependencies();
+  const auto& output_expressions = this->output_expressions();
+  for (const auto& input_order_dependency : input_order_dependencies) {
+    if (!(contains_all_expressions(input_order_dependency.ordering_expressions, output_expressions) &&
+          contains_all_expressions(input_order_dependency.ordered_expressions, output_expressions))) {
+      continue;
+    }
+    order_dependencies.emplace(input_order_dependency);
+  }
+
+  return order_dependencies;
```

Comment:
Think we discussed this offline and we basically had originally sets, the pointers, then sets again (https://github.com/hyrise/hyrise/pull/2528#discussion_r1091075056). I do not think this makes a difference now, since we do not request dependencies often.